### DISCUSSION TITLE
Exposure of managed functions to blueprints

### DISCUSSION THREAD

**Discussion by TheEndHunter** [nxrighthere/UnrealCLR#76] [Help] [answered]
Hi.
I have just started using UnrealCLR for the first time today and after setting up i was starting to build this function:

here is the code in c#:
![chunk](https://user-images.githubusercontent.com/8478343/102627627-5aae0b00-4140-11eb-97ac-95c402fb1585.png)
here's the blueprint:
![blueprint](https://user-images.githubusercontent.com/8478343/102627963-e0ca5180-4140-11eb-81ce-4a02e41c21a5.png)

base on the above i would like to find out:
A) why does this blueprint fail to find the managed code?
B) How do i pass variables for the function to the c# code?
C) How to i return values back to the blueprints?

Thankyou in advanced :)

**Comment by nxrighthere** [comment, marked as answer]
The managed function is ignored by the plugin because it doesn't match the arguments in blueprints. At the moment, it's possible to pass to the function only an object which might have properties as parameters, but it's not possible to pass custom arguments (this is work-in-progress, see https://github.com/nxrighthere/UnrealCLR/issues/33).

To get an idea of how to pass an object, see `TestBlueprintActorFunction()` and `TestBlueprintComponentFunction()` in [this](https://github.com/nxrighthere/UnrealCLR/blob/master/Source/Managed/Tests/BlueprintsExtensibility.cs) test. Optionally, managed functions also might have no arguments.

To pass data back to blueprints, you can use event dispatchers.

**Reply by TheEndHunter** [reply]
thanks! ill take a look
